### DISCUSSION TITLE
Consider nesting `Calibration` in `Device`

### DISCUSSION THREAD

**Discussion by bruno-f-cruz** [AllenNeuralDynamics/aind-data-schema#855] [Ideas]
**Is your feature request related to a problem? Please describe.**
In all the cases I have encountered, `Calibration` is always tied to a specific device. Indeed, the current spec makes this clear in the `Calibration` class by adding a `device_name` REQUIRED property to keep track of this association
https://github.com/AllenNeuralDynamics/aind-data-schema/blob/0328cbed69d5da849660557be162a24c2cd02df4/src/aind_data_schema/models/devices.py#L270-L278

**Describe the solution you'd like**
Make calibration a nested field inside `Device` with the `Optional[Calibration]` typing. For hardware instantiation and experiment specification, this would be a huge help and it would allow users to not have to grab cross-referenced data across different classes (well actually across schemas, since Calibration is currently also stored in Session (?))

**Describe alternatives you've considered**
N/A

**Comment by dbirman** [comment]
I think the current system is working as-intended (rig calibrations go in rig and per-session calibrations go in session) and there's no plan right now to change it for 2.0. I'm going to close this but feel free to re-open if you want to discuss more or find me in person.
